docs/lang/articles/contribution/dev_install.md

```diff
@@ -4,318 +4,467 @@ sidebar_position: 2
-     export PATH=<path_to_llvm_folder>/bin:$PATH
-     ```
+This installation guide is *NOT* intended for end users who only wish to do simulation or high performance numerical computation. We recommend that end users install Taichi via `pip install taichi` and that there is no need for you to build Taichi from source. Doing both at the same time causes unnecessary conflicts. 
 
-   - On other Linux distributions, please search [this site](https://pkgs.org) for clang version \>= 8.
-   - On Windows: Please download [clang-10](https://github.com/taichi-dev/taichi_assets/releases/download/llvm10/clang-10.0.0-win.zip). Make sure you add the `bin` folder containing `clang.exe` to the `PATH` environment variable.
+See the [Get Started](https://docs.taichi.graphics/) for more information on quickly setting up Taichi for end users. 
 
-:::note
-On Linux, `clang` is the **only** supported compiler for compiling the Taichi package.
 :::
 
-:::danger
-Taichi doesn't support building with `clang++-12` and above yet.
-`clang++-10` is highly recommended but if you're on a newer (>=11) macOS,
-building with `clang++-11` also works through `brew install llvm@11` and `export CXX=/opt/homebrew/opt/llvm@11/bin/clang++`.
-:::
+## Introduction
+
+  This installation guide covers the following:
+
+  - Prerequisites for building Taichi from source
+  - Building Taichi from source
+  - Installing optional dependencies
+  - Troubleshooting and debugging
+  - Frequently asked questions
 
 :::note
-On Linux, some additional packages might be required to build Taichi. E.g., on Ubuntu 20.04, you may need `libxi-dev` `libxcursor-dev` `libxinerama-dev` `libxrandr-dev` `libx11-dev` `libgl-dev`. please check the output of of CMake when building from source.
+
+Installation instructions vary depending on which operating system (OS) you are using. Choose the right OS or platform before you proceed.
+
 :::
 
-3. LLVM: Make sure you have version 10.0.0 installed. Taichi uses a **customized LLVM**, which we provided as binaries depending on your system environment. Note that the pre-built binaries from the LLVM official website or other sources may not work.
-   - [LLVM 10.0.0 for Linux](https://github.com/taichi-dev/taichi_assets/releases/download/llvm10_linux_patch2/taichi-llvm-10.0.0-linux.zip)
-   - [LLVM 10.0.0 for macOS (excluding M1)](https://github.com/taichi-dev/taichi_assets/releases/download/llvm10/taichi-llvm-10.0.0-macos.zip)
-   - [LLVM 10.0.0 for macOS (M1 chip)](https://github.com/taichi-dev/taichi_assets/releases/download/llvm10_m1/llvm-10.0.0-m1.zip)
-   - [LLVM 10.0.0 for Windows MSVC 2019](https://github.com/taichi-dev/taichi_assets/releases/download/llvm10/taichi-llvm-10.0.0-msvc2019.zip)
+## Prerequisites
 
+|              OS              |                                                                  Windows/macOS/Ubuntu/Arch Linux/Other Linux distributions                                                                  |
```

Comment:
I don't think `OS` should be the table header. This row seems parallel to the following items.

Reply:
Agreed.